azure_functions_worker/dispatcher.py

```diff
@@ -327,24 +329,29 @@ async def _handle__worker_status_request(self, request):
     async def _handle__functions_metadata_request(self, request):
         metadata_request = request.functions_metadata_request
         directory = metadata_request.function_app_directory
-        function_path = os.path.join(directory, SCRIPT_FILE_NAME)
+        script_file_name = get_app_setting(
+            setting=PYTHON_SCRIPT_FILE_NAME,
+            default_value=f'{PYTHON_SCRIPT_FILE_NAME_DEFAULT}')
+        function_path = os.path.join(directory, script_file_name)
 
         logger.info(
-            'Received WorkerMetadataRequest, request ID %s, directory: %s',
-            self.request_id, directory)
-
-        if not os.path.exists(function_path):
-            # Fallback to legacy model
-            logger.info("%s does not exist. "
-                        "Switching to host indexing.", SCRIPT_FILE_NAME)
-            return protos.StreamingMessage(
-                request_id=request.request_id,
-                function_metadata_response=protos.FunctionMetadataResponse(
-                    use_default_metadata_indexing=True,
-                    result=protos.StatusResult(
-                        status=protos.StatusResult.Success)))
+            'Received WorkerMetadataRequest, request ID %s, function_path: %s',
+            self.request_id, function_path)
 
         try:
+            validate_script_file_name(script_file_name)
+
+            if not os.path.exists(function_path):
+                # Fallback to legacy model
+                logger.info("%s does not exist. "
```

Comment:
Is this necessary to log in general? If this log can be merged in the earlier log itself?